docs/core/diagnostics/trace-perfcollect-lttng.md

```diff
@@ -0,0 +1,201 @@
+   > mkdir helloWorld
+   > cd helloWorld
+   > dotnet new console
+   > dotnet publish --self-contained -r linux-x64
+   >```
+
+This creates a new Hello World application and builds it as a self-contained app. Note that if you have multiple versions of the .NET Runtime installed, the instructions above will use the latest. As long as your app also uses the latest (likely) then these instructions will work without modification.
+
+As a side effect of creating the self-contained application the dotnet tool will download a nuget package called runtime.linux-x64.microsoft.netcore.app and place it in the directory ~/.nuget/packages/runtime.linux-x64.microsoft.netcore.app/VERSION, where VERSION is the version number of your .NET Core runtime (e.g. 2.1.0). Under that is a tools directory and inside there is the crossgen tool you need. Starting with .NET Core 3.0, the package location is ~/.nuget/packages/microsoft.netcore.app.runtime.linux-x64/VERSION.
+
+The `crossgen` tool needs to be put next to the runtime that is actually used by your application. Typically your app uses the shared version of .NET Core that is installed at /usr/share/dotnet/shared/Microsoft.NETCore.App/VERSION where VERSION is the version number of the .NET Runtime. This is a shared location, so you need to be super-user to modify it. If the VERSION is 2.1.0 the commands to update `crossgen` would be:
+
+   >```bash
+   > sudo bash
+   > cp ~/.nuget/packages/runtime.linux-x64.microsoft.netcore.app/2.1.0/tools/crossgen /usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.0
+   >```
+
+Once you have done this, `perfcollect` will use crossgen to include framework symbols. The warning that `perfcollect` used to issue should go away. This only has to be one once per machine (until you update your runtime).
+
+### Alternative: Turn off use of precompiled code
+
+If you don't have the ability to update the .NET Runtime (to add `crossgen`), or if the above procedure did not work for some reason, there is another approach to getting framework symbols. You can tell the runtime to simply not use the precompiled framework code. The code will be Just-In-Time compiled and the `crossgen` tool is not needed.
+
+> [!NOTE]
+> Choosing this approach may increase the startup time for your application.
+
+To do this, you can add the following environment variable:
+
+```bash
+export COMPlus_ZapDisable=1
+```
+
+With this change you should get the symbols for all .NET code.
+
+## Getting Symbols For the Native Runtime
+
+Most of the time you are interested in your own code, which `perfcollect` resolves by default. Sometimes it is very useful to see what is going on inside the .NET DLLs (which is what the last section was about), but sometimes what is going on in the native runtime dlls (typically libcoreclr.so), is interesting.  `perfcollect` will resolve the symbols for these when it converts its data, but only if the symbols for these native DLLs are present (and are beside the library they are for).
```

Comment:
Similar to the framework symbols, I would put a line of text+link in chronological order within the workflow that tells users they need to address this concern before they take the trace.

@brianrob - would it be hard to get perfcollect to do this automatically in the future?

Reply:
This would not be hard to do.  In most cases, we get this for free.  The specific cases where this doesn't work is when the binary gets stripped - I believe this only happens for distro-specific packages (e.g. deb, rpm)  It would not be hard to detect whether or not we should download the symbols at collection time.